File: Day 38-39/diceBattle.js
function random(min, max){

    var res;
    res = Math.floor(Math.random() * (max - min +1)) + min;
    return res;

}

function throwDice(faces, times){

    let rolls = []

    for(let i=0; i<times; i++ ){

        let number = random(1, faces)

        rolls.push(number)

    }

    return rolls

}


function battle(rolls){

    let rolls1 = throwDice(6, rolls) 
    let rolls2 = throwDice(6, rolls) 
    let rolls3 = throwDice(6, rolls) 

    const reducer = (acum, rolls) => acum+rolls

    let player1 = rolls1.reduce( reducer , 0 )
    let player2 = rolls2.reduce( reducer , 0 )
    let player3 = rolls3.reduce( reducer , 0 )

    console.log(`player1 rolls: ${rolls1} => score 1: ${player1}`)
    console.log(`player2 points: ${rolls2} => score 2: ${player2}`)
    console.log(`player3 points: ${rolls3} => score 3: ${player3}`)

    
    if( player1 === player2 && player1 === player3 )
        console.log('Triple tie!')
    else if( player1 === player2 && player1 > player3 )
        console.log('Player 3 lose!')
    else if( player1 === player3 && player1 > player2 )
        console.log('Player 2 lose!')
    else if( player3 === player2 && player3 > player1)
        console.log('Player 1 lose!')
    else if( player1 > player2 && player1 > player3 )
        console.log('Player 1 win!')
    else if( player2 > player3 && player2 > player1 )
        console.log('Player 2 win!')
    else if( player3 > player2 && player3 > player1 )
        console.log('Player 3 win!')

}


battle(2)
battle(5)

export default throwDice

File: Day 38-39/diceBattle.c
#include <stdio.h>
#include <math.h>

int * throwDice(int faces, int times);
void diceBattle(int rolls);
int arrayTotal(int * array, int len);

int main()
{

    diceBattle(2);
    diceBattle(5);
    diceBattle(8);
    diceBattle(10);

    return 0;

}

int * throwDice(int faces, int times){

    int rolls[times];

    int i;
    for( i=0; i<times; i++){
        int roll = rand()%faces + 1;
        rolls[i] = roll;
        printf("%d, ", rolls[i]);
    }
    printf("\n");

    int * rolls_ = rolls;
    return rolls_;

}

int arrayTotal(int * array, int len){

    int total = 0;

    int i;
    for( i=0; i<len; i++ ){
        total += array[i];
    }

    return total;

}

void diceBattle(int rolls){

    int * rolls1 = throwDice(6, rolls);
    int player1 = arrayTotal(rolls1, rolls);

    int * rolls2 = throwDice(6, rolls);
    int player2 = arrayTotal(rolls2, rolls);
    
    int * rolls3 = throwDice(6, rolls); 
    int player3 = arrayTotal(rolls3, rolls);

    printf("%d - %d - %d \n", player1, player2, player3);
    
    if( player1 == player2 && player1 == player3 )
        printf("Triple tie! \n");
    else if( player1 == player2 && player1 > player3 )
        printf("Player 3 lose! \n");
    else if( player1 == player3 && player1 > player2 )
        printf("Player 2 lose! \n");
    else if( player3 == player2 && player3 > player1)
        printf("Player 1 lose! \n");
    else if( player1 > player2 && player1 > player3 )
        printf("Player 1 win! \n");
    else if( player2 > player3 && player2 > player1 )
        printf("Player 2 win! \n");
    else if( player3 > player2 && player3 > player1 )
        printf("Player 3 win! \n");

}